qiskit_nature/operators/second_quantization/star_algebra.py

```diff
@@ -0,0 +1,86 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Sum Operator base interface."""
+
+from abc import ABC, abstractmethod
+
+from .multiply import MultiplyMixin
+from .tolerances import TolerancesMixin
+
+
+class StarAlgebraMixin(MultiplyMixin, TolerancesMixin, ABC):
+    """The star algebra mixin class.
+
+    Star algebra means an algebra with adjoint (dagger).
+    """
+
+    @abstractmethod
+    def _add(self, other):
+        r""" Return Operator addition of self and other, overloaded by ``+``.
+        """
+        raise NotImplementedError
+
+    def __add__(self, other):
+        return self._add(other)
+
+    def __sub__(self, other):
+        return self._add(other._multiply(-1))
+
+    def __mul__(self, other: complex):
```

Comment:
Why does this have a type on it. From what I can tell for add etc the type would end up being defined by the class using the mixin.

Also should we not have `__rmul__` etc here

Reply:
Thanks. You are right. This type hint is unnecessary.

quantum_info's multiplication is a bit strange from fermionic_op and opflow. `*` is for scalar multiplication and dot, `@` is for compose(). Note that this compose is a product of opposite direction. (i.e. A * B = B @ A) This property is incompatible for opflow and seocnd_quantized_op, so I made StarSlgebraMixin. After this PR, I am planning to move this mixin to opflow as a part of opflow refactoring.